src/main/java/io/github/cottonmc/cotton/datapack/recipe/CookingWithBonusRecipe.java

```diff
@@ -0,0 +1,120 @@
+package io.github.cottonmc.cotton.datapack.recipe;
+
+import com.google.gson.JsonObject;
+import net.minecraft.item.ItemStack;
+import net.minecraft.recipe.Ingredient;
+import net.minecraft.recipe.RecipeSerializer;
+import net.minecraft.recipe.RecipeType;
+import net.minecraft.recipe.cooking.CookingRecipe;
+import net.minecraft.recipe.crafting.ShapedRecipe;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.PacketByteBuf;
+import net.minecraft.world.loot.LootManager;
+import net.minecraft.world.loot.context.LootContext;
+
+import java.util.Collections;
+import java.util.List;
+
+import static net.minecraft.util.JsonHelper.*;
+
+/**
+ * Implements a generic machine processing recipe with optional loot table for bonus items.
+ *
+ * Example JSON:
+ *
+ * {
+ *     "type": "c:example",
+ *     "ingredient": {
+ *         "item": "minecraft:iron_ore"
+ *     },
+ *     "result": {
+ *         "item": "example:iron_dust",
+ *         "count": 2
+ *     },
+ *     "bonus_loot_table": "example:iron_ore_to_dust_bonuses"
+ * }
+ *
+ * @see CrushingRecipe for an example implementation
+ */
+public abstract class CookingWithBonusRecipe extends CookingRecipe {
```

Comment:
Not sure it's the best idea to extend CookingRecipe, due to how recipes are shown in the recipe book. I think it'd be better to reimpl the things cooking recipe has in it.

Reply:
Sure